docs/apache-airflow/usage-cli.rst

```diff
@@ -199,3 +199,52 @@ Both ``json`` and ``yaml`` formats make it easier to manipulate the data using c
     "sd": "2020-11-29T14:53:56.931243+00:00",
     "ed": "2020-11-29T14:53:57.126306+00:00"
   }
+
+.. _cli-db-clean:
+
+Purge history from metadata database
+------------------------------------
+
+.. note::
+
+  It's strongly recommended that you backup the metadata database before running the ``db clean`` command.
+
+The ``db clean`` command works by deleting from each table the records older than the provided ``--clean-before-timestamp``.
+
+You can optionally provide a list of tables to perform deletes on. If no list of tables is supplied, all tables will be included.
+
+You can use the ``--dry-run`` option to print the row counts in the primary tables to be cleaned.
+
+Beware cascading deletes
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Keep in mind that some tables have foreign key relationships defined with ``ON DELETE CASCADE`` so deletes in one table may trigger deletes in others.  For example, the ``task_instance`` table keys to the ``dag_run`` table, so if a dag run record is deleted, all of its associated task instances will also be deleted.
+
+
+Special handling for dag runs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Commonly, Airflow determines which DagRun to run next by looking up the latest DagRun.  If you delete all dag runs, Airflow may schedule an old dag run that was already completed, e.g. if you have set ``catchup=True``.  So the ``db clean`` will preserve the latest non-manually-triggered dag run to preserve continuity in scheduling.
+
+Considerations for backfillable dags
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Not all dags are designed for use with airflow's backfill command.  But for those which are, special care is warranted.  If you delete dag runs, and if you run backfill over a range of dates that includes the deleted dag runs, those runs will be recreated and run again.  For this reason, if you have dags that fall into this category you may want to refrain from deleting dag runs and only clean other large tables such as task instance and log etc.
+
+
+.. _cli-db-downgrade:
+
+Downgrading airflow
+-------------------
+
+.. note::
+
+    It's recommended that you backup your database before running ``db downgrade`` or any other database operation.
+
+You can downgrade to a particular Airflow version with the ``db downgrade`` command.  Alternatively you may provide an alembic revision id to downgrade to.
+
+If you want to preview the commands but not execute them, use option ``--sql-only``.
+
+Options ``--from-revision`` and ``--from-version`` may only be used in conjunction with the ``--sql-only`` option, because if actually *running* migrations we should always downgrade from current revision.
```

Comment:
```suggestion
Options ``--from-revision`` and ``--from-version`` may only be used in conjunction with the ``--sql-only`` option, because when actually *running* migrations we should always downgrade from current revision.
```
Maybe?